### PR TITLE
Allow update to take a reducer.

### PR DESCRIPTION
**Pull request by jbucaran**
A reducer is a function that takes the state and returns a new state or a reducer.
    
```js
update(state => {
  return { value: state.value + 1 }
}
```    
    
This allows you to get the current & latest state inside the callback of some async call.
    
The alternative is using the state that was passed to the action that originated the async call. The problem is that this reference to the state may have gone out-of-date if other actions run and finished during the async process and changed the state.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,7 @@ yarn.lock
 package-lock.json
 
 # Misc
-sandbox
+example
```
Comment (okwolf) [inline .gitignore]: We're not using either of these are we?

Comment (jbucaran) [reply .gitignore]: @okwolf I am locally, but it's not committed to the repo.

**Comment by jbucaran** [comment]
Updated docs and added 100% coverage.

I'll merge here and now work on a substantial docs update.

**Comment by jbucaran** [comment]
Related https://github.com/zaceno/hyperapp-partial/issues/3

**Pull request merged**